<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="wei wang" />

<meta name="date" content="2017-02-09" />

<title>HumanTungiPSC missing pattern analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic flash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/NKweiwang/logisticFlash_workflow">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">HumanTungiPSC missing pattern analysis</h1>
<h4 class="author"><em>wei wang</em></h4>
<h4 class="date"><em>2017-02-09</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-10</p>
<p><strong>Code version:</strong> ab42775</p>
<pre class="r"><code># Biobase is for accessing expressionSet objects
library(Biobase)

# Load Buettner data
devtools::install_github(&quot;jhsiao999/singleCellRNASeqHumanTungiPSC&quot;)
library(singleCellRNASeqHumanTungiPSC)

# Extract expression data
eset &lt;- get(data(&quot;HumanTungiPSC&quot;))

# Access the count matrix
counts &lt;- exprs(eset)
dim(counts)</code></pre>
<pre><code>[1] 12235   560</code></pre>
<pre class="r"><code>Binary_count = 1 * (counts != 0)
Boolean_count = 2*( Binary_count - 1/2)
# Access cell sample information 
#pData(eset)
#cluster &lt;- pData(eset)$Cluster
#table(cluster)
barlabels = pData(HumanTungiPSC)
total_counts = apply(counts,2,sum)</code></pre>
<div id="run-logistic-flash-on-the-missing-position" class="section level2">
<h2>run Logistic Flash on the missing position</h2>
<p>We run the Logistic flash with greedy algorithm</p>
<p>Before run the algorithm, we have some guess or hypothesis on the missing pattern</p>
<div id="hypothesis" class="section level3">
<h3>hypothesis</h3>
<ol style="list-style-type: decimal">
<li>the first factor should be correlated with the total counts of each cell type, which represent the total missing pattern.</li>
<li>first factor(total missing pattern) should be correlated with total counts.</li>
<li>there are two types of missing patterns besides the first factor (total missing)</li>
</ol>
<ul>
<li>“technical missing”</li>
<li>“system missing”</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>“technical missing” should be correlated with individual and bacth.</li>
<li>“system missing” should be correlated with cell cycle and total counts.</li>
<li>“technical missing” should not be correlated with the cell cycle (or total count)</li>
</ol>
<ul>
<li>total counts is column sum of the raw counts data</li>
</ul>
<pre class="r"><code>source(&#39;~/HG/LogisticFlash/Rcode/GD_Rfuncrtions.R&#39;)
# MissPattern = GL_flash(Boolean_count,K = 20)
# saveRDS(MissPattern,file = &quot;~/HG/LogisticFlash/data/MissPatternHumanTungiPSC_K20.rds&quot;)
MissPattern = readRDS(&quot;~/HG/LogisticFlash/data/MissPatternHumanTungiPSC_K20.rds&quot;)</code></pre>
</div>
</div>
<div id="missing-pattern-from-logistic-flash" class="section level2">
<h2>Missing Pattern from Logistic Flash</h2>
<p>In the following result, we find that</p>
<ol style="list-style-type: decimal">
<li>factor 1 is the total missing pattern, which represent the total missing for each cell type.</li>
<li>factor 1 is correlated with total counts.</li>
<li>factor 2 is correlated with total counts.</li>
<li>all other factors (factor3 - factor8) has nothing to do with total counts.</li>
<li>factor 3 - factor 8 reflect the information of individual and batch.</li>
</ol>
<p>we can guess that:</p>
<ol style="list-style-type: decimal">
<li>factor one is total missing pattern</li>
<li>factor two is “system missing” (“biological missing”)</li>
<li>factor three - factor eight are “technical missing”</li>
</ol>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-64-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-64-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-64-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-64-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="individual-information-on-factors" class="section level3">
<h3>individual information on factors</h3>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-65-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-cycle-information-on-factor" class="section level3">
<h3>cell cycle information on factor</h3>
<pre class="r"><code>library(ggplot2)
p &lt;- ggplot(plot_df, aes(total_count, f2, label = (as.character(phase) )))
# p + geom_text()
p + geom_label(aes(fill = factor(phase)), colour = &quot;white&quot;, fontface = &quot;bold&quot;)</code></pre>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>numeric_phase = phase
numeric_phase[which(phase == &quot;G1.S&quot;)] = &quot;1&quot;
numeric_phase[which(phase == &quot;S&quot;)] = &quot;2&quot;
numeric_phase[which(phase == &quot;G2&quot;)] = &quot;3&quot;
numeric_phase[which(phase == &quot;G2.M&quot;)] = &quot;4&quot;
numeric_phase[which(phase == &quot;M.G1&quot;)] = &quot;5&quot;
plot(as.numeric(as.factor(phase)),MissPattern$f[,2],type = &quot;p&quot;, main = &quot;cell cycle phase vs factor 2&quot;)</code></pre>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-67-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(as.numeric(as.factor(phase)),plot_df$total_count, type = &quot;p&quot;, main = &quot;cell cycle phase vs total count&quot;)</code></pre>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-67-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="point-the-cell-cycle-gene-into-the-factor-2-this-might-be-not-correct" class="section level3">
<h3>point the cell cycle gene into the factor 2 (this might be not correct)</h3>
<pre class="r"><code>plot(MissPattern$l[,2],pch = 16)
G1.S_index = which(row.names(counts) %in% cellcycle_genes$G1.S)
S_index = which(row.names(counts) %in% cellcycle_genes$S)
G2_index = which(row.names(counts) %in% cellcycle_genes$G2)
G2.M_index = which(row.names(counts) %in% cellcycle_genes$G2.M)
M.G1_index = which(row.names(counts) %in% cellcycle_genes$M.G1)
points(G1.S_index, MissPattern$l[G1.S_index,2], pch = 16, col = &quot;red&quot;)
points(S_index, MissPattern$l[S_index,2], pch = 16, col = &quot;yellow&quot;)
points(G2_index, MissPattern$l[G2_index,2], pch = 16, col = &quot;skyblue&quot;)
points(G2.M_index, MissPattern$l[G2.M_index,2], pch = 16, col = &quot;blue&quot;)
points(M.G1_index, MissPattern$l[M.G1_index,2], pch = 16, col = &quot;green&quot;)</code></pre>
<p><img src="figure/HumanTungiPSC.Rmd/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.0 (2016-05-03)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.12.3 (unknown)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] biomaRt_2.24.1                      
[2] ggplot2_2.2.1                       
[3] MASS_7.3-45                         
[4] singleCellRNASeqHumanTungiPSC_0.99.0
[5] Biobase_2.28.0                      
[6] BiocGenerics_0.14.0                 
[7] workflowr_0.3.0                     
[8] rmarkdown_1.3                       

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.9          GenomeInfoDb_1.4.3   plyr_1.8.4          
 [4] git2r_0.18.0         bitops_1.0-6         iterators_1.0.8     
 [7] tools_3.3.0          digest_0.6.12        RSQLite_1.1-2       
[10] tibble_1.2           evaluate_0.10        memoise_1.0.0       
[13] gtable_0.2.0         lattice_0.20-34      Matrix_1.2-8        
[16] foreach_1.4.3        DBI_0.5-1            rstudioapi_0.6      
[19] curl_2.3             yaml_2.1.14          withr_1.0.2         
[22] httr_1.2.1           stringr_1.1.0        knitr_1.15.1        
[25] IRanges_2.2.9        S4Vectors_0.6.6      devtools_1.12.0     
[28] REBayes_0.73         stats4_3.3.0         rprojroot_1.2       
[31] grid_3.3.0           R6_2.2.0             AnnotationDbi_1.30.1
[34] XML_3.98-1.5         irlba_2.1.2          flashr_0.1.1        
[37] ashr_2.0.5           magrittr_1.5         scales_0.4.1        
[40] backports_1.0.5      codetools_0.2-15     htmltools_0.3.5     
[43] assertthat_0.1       colorspace_1.3-2     labeling_0.3        
[46] stringi_1.1.2        Rmosek_7.1.2         RCurl_1.95-4.8      
[49] lazyeval_0.2.0       munsell_0.4.3        doParallel_1.0.10   
[52] pscl_1.4.9           truncnorm_1.0-7      SQUAREM_2016.8-2    </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
