---
title: "HumanTungiPSC missing pattern analysis"
author: "wei wang"
date: 2017-02-09
output: html_document
---

```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

```{r knitr-opts-chunk, include=FALSE}
```

**Last updated:** `r Sys.Date()`

**Code version:** `r workflowr::extract_commit(".", 1)$sha1`


```{r,warning=FALSE,cache=TRUE,eval=TRUE,message=FALSE}
# Biobase is for accessing expressionSet objects
library(Biobase)

# Load Buettner data
devtools::install_github("jhsiao999/singleCellRNASeqHumanTungiPSC")
library(singleCellRNASeqHumanTungiPSC)

# Extract expression data
eset <- get(data("HumanTungiPSC"))

# Access the count matrix
counts <- exprs(eset)
dim(counts)

Binary_count = 1 * (counts != 0)
Boolean_count = 2*( Binary_count - 1/2)
# Access cell sample information 
#pData(eset)
#cluster <- pData(eset)$Cluster
#table(cluster)
barlabels = pData(HumanTungiPSC)
total_counts = apply(counts,2,sum)
```


## run Logistic Flash on the missing position

We run the Logistic flash with greedy algorithm

Before run the algorithm, we have some guess or hypothesis on the missing pattern

### hypothesis

1. the first factor should be correlated with the total counts of each cell type, which represent the total missing pattern.
2. first factor(total missing pattern) should be correlated with total counts.
3. there are two types of missing patterns besides the first factor (total missing)
  + "technical missing"
  + "system missing"
4. "technical missing" should be correlated with individual and bacth.
5. "system missing" should be correlated with cell cycle and total counts.
6. "technical missing" should not be correlated with the cell cycle (or total count)


- total counts is column sum of the raw counts data


```{r,warning=FALSE,cache=TRUE,eval=TRUE,message=FALSE}
source('~/HG/LogisticFlash/Rcode/GD_Rfuncrtions.R')
# MissPattern = GL_flash(Boolean_count,K = 20)
# saveRDS(MissPattern,file = "~/HG/LogisticFlash/data/MissPatternHumanTungiPSC_K20.rds")
MissPattern = readRDS("~/HG/LogisticFlash/data/MissPatternHumanTungiPSC_K20.rds")
```


## Missing Pattern from Logistic Flash

In the following result, we find that

1. factor 1 is the total missing pattern, which represent the total missing for each cell type. 
2. factor 1 is correlated with total counts.
3. factor 2 is correlated with total counts.
4. all other factors (factor3 - factor8) has nothing to do with total counts.
5. factor 3 - factor 8 reflect the information of individual and batch.

we can guess that:

1. factor one is total missing pattern
2. factor two is "system missing" ("biological missing")
3. factor three - factor eight are "technical missing"


```{r,echo=FALSE}
par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
for( i in 1:8) barplot(MissPattern$f[,i],main = paste("factor",i))
total_counts = apply(counts,2,sum)
for( i in 1:8) plot(total_counts,MissPattern$f[,i],main = paste("factor",i,"vs","total counts"))
```


### individual information on factors

```{r,echo=FALSE}
library(ggplot2)
plot_df = data.frame(MissPattern$f)
colnames(plot_df) = c("f1","f2","f3","f4","f5","f6","f7","f8")
plot_df$total_count = total_counts
# scatter plot
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p1 <- ggplot(plot_df, aes(total_count, f1, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p1 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")

par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p2 <- ggplot(plot_df, aes(total_count, f2, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p2 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p3 <- ggplot(plot_df, aes(total_count, f3, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p3 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p4 <- ggplot(plot_df, aes(total_count, f4, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p4 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p5 <- ggplot(plot_df, aes(total_count, f5, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p5 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p6 <- ggplot(plot_df, aes(total_count, f6, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p6 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p7 <- ggplot(plot_df, aes(total_count, f7, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p7 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")
  
  par(mfrow = c(2,2),mar = c(5,4,4,2) - 1.6)
  p8 <- ggplot(plot_df, aes(total_count, f8, label = (as.character(factor(barlabels$individual )) )), parse = FALSE)
  # p + geom_text()
  p8 + geom_label(aes(fill = factor(barlabels$individual)), colour = "white", fontface = "bold")

```


### cell cycle information on factor

```{r,echo=FALSE}
# This function estimates cell cycle phase 
# The method is adapted from Whitefield et al.
cellcycle_phase_score <- function(cellcycle_geneslist, data_expression) {

  phase_score <- sapply(cellcycle_geneslist, function(xx){
    
    ## make gene by cell tables for each cell cycle phase
    expression_eachphase <- data_expression[rownames(data_expression) %in% unlist(xx) ,]
  
    ## add to each table the average expression of phase-specific genes
    combined_matrix <- rbind(expression_eachphase, average = apply(data_expression,2,mean) )
    
    ## compute correlation between gene expression and phase-specific average gene expression
    cor_matrix <- cor(t(combined_matrix))
    cor_vector <- cor_matrix[,dim(cor_matrix)[1]]
    
    ## restrict to correlation >= 0.3 
    expression_eachphase_restricted <- expression_eachphase[rownames(expression_eachphase) %in% names(cor_vector[cor_vector >= 0.3]),]
    
    ## output phase specific scores (mean of normalized expression levels in the phase)
    apply(expression_eachphase, 2, mean)
  })

  ## Two-step normalization (by gene and by cell)
  ## by row (gene)
  gene_mean <- apply(phase_score, 1, mean)
  gene_sd   <- apply(phase_score, 1, sd)
  phase_score_gene_normed <- do.call(rbind, 
                                lapply(1:dim(phase_score)[1], function(i) {
                                    (phase_score[i,] - gene_mean[i])/gene_sd[i]
                                })  )
  
  ## by column (cell)
  cell_mean <- apply(phase_score_gene_normed, 2, mean)
  cell_sd   <- apply(phase_score_gene_normed, 2, sd)
  score_final_normed <- do.call(cbind, 
                            lapply(1:dim(phase_score_gene_normed)[2], function(i) {
                                (phase_score_gene_normed[, i] - cell_mean[i])/cell_sd[i]
                            }) )
  return(score_final_normed)
}

cellcycle_genes <- read.table("~/HG/LogisticFlash_workflow/data/cellcyclegenes.txt",
                               header = TRUE, sep = "\t",
                               stringsAsFactors = FALSE)
colnames(cellcycle_genes)[colnames(cellcycle_genes) == "G2.M"] <- "G2"
colnames(cellcycle_genes)[colnames(cellcycle_genes) == "M"] <- "G2.M"
phase <- apply( cellcycle_phase_score(cellcycle_genes, counts), 
                       1, function(x) colnames(cellcycle_genes)[which.max(x)])
```


```{r}
library(ggplot2)
p <- ggplot(plot_df, aes(total_count, f2, label = (as.character(phase) )))
# p + geom_text()
p + geom_label(aes(fill = factor(phase)), colour = "white", fontface = "bold")
numeric_phase = phase
numeric_phase[which(phase == "G1.S")] = "1"
numeric_phase[which(phase == "S")] = "2"
numeric_phase[which(phase == "G2")] = "3"
numeric_phase[which(phase == "G2.M")] = "4"
numeric_phase[which(phase == "M.G1")] = "5"
plot(as.numeric(as.factor(phase)),MissPattern$f[,2],type = "p", main = "cell cycle phase vs factor 2")
plot(as.numeric(as.factor(phase)),plot_df$total_count, type = "p", main = "cell cycle phase vs total count")
```

### point the cell cycle gene into the factor 2 (this might be not correct)

```{r}
plot(MissPattern$l[,2],pch = 16)
G1.S_index = which(row.names(counts) %in% cellcycle_genes$G1.S)
S_index = which(row.names(counts) %in% cellcycle_genes$S)
G2_index = which(row.names(counts) %in% cellcycle_genes$G2)
G2.M_index = which(row.names(counts) %in% cellcycle_genes$G2.M)
M.G1_index = which(row.names(counts) %in% cellcycle_genes$M.G1)
points(G1.S_index, MissPattern$l[G1.S_index,2], pch = 16, col = "red")
points(S_index, MissPattern$l[S_index,2], pch = 16, col = "yellow")
points(G2_index, MissPattern$l[G2_index,2], pch = 16, col = "skyblue")
points(G2.M_index, MissPattern$l[G2.M_index,2], pch = 16, col = "blue")
points(M.G1_index, MissPattern$l[M.G1_index,2], pch = 16, col = "green")
```

## Session Information

```{r session-info}
```
